pinot-tools/src/main/java/com/linkedin/pinot/tools/PinotToolLauncher.java

```diff
@@ -35,6 +35,8 @@
   @Argument(handler = SubCommandHandler.class, metaVar = "<subCommand>")
   @SubCommands({
       @SubCommand(name = "UpdateSegmentState", impl = UpdateSegmentState.class),
+      @SubCommand(name = "PurgeSegmentsFromFiler", impl = FilerSegmentPurger.class),
```

Comment:
the code needs to be elsewhere in that case. Pinot-controller can't share tools, IIRC.

File name has no bearing on the actual segment name in Pinot. I know it doesn't happen but it's possible to upload tar file with name different from segment name.